bns/getKoboData.js

```diff
@@ -20,15 +25,35 @@ each(dataPath('surveys[*]'), state => {
         body: submission,
       };
     });
-    const count = state.data.submissions.length;
+
+    state.surveySubmissions.push(...results);
+    const count = results.length;
     console.log(`Fetched ${count} submissions from ${formId} (${tag}).`);
     //Once we fetch the data, we want to post each individual Kobo survey
     //back to the OpenFn inbox to run through the jobs =========================
-    return each(dataPath('submissions[*]'), state => {
-      console.log(`Posting ${state.data.i + 1} of ${count}...`);
-      return post(state.configuration.openfnInboxUrl, {
-        body: state => state.data,
-      })(state);
-    })(state);
-  })(state);
-});
\ No newline at end of file
+  })(state)
+    .then(() => {
```

Comment:
@hunterachieng why not `.then((state) => {return state})` ?

Reply:
Actually since we are using the `.then()` approach i think we can remove the callback completely and have something like this 

```
return get(`${url}${query}`)(state).then((state) => {
  const results = state.data.results.map((submission, i) => {
    return {
      i,
      // Here we append the tags defined above to the Kobo form submission data
      form: tag,
      formName: name,
      formOwner: owner,
      body: submission,
    };
  });

  state.surveySubmissions.push(...results);
  const count = results.length;
  console.log(`Fetched ${count} submissions from ${formId} (${tag}).`);

  return state;
});
```